### PR TITLE
docs: Correct default values

### PR DESCRIPTION
**Pull request by jaxvanyang**
Some default values in the doc are outdated.

Release Notes:

- N/A

### CODE REVIEW

**Comment by jaxvanyang** [comment]
@cla-bot check

File: docs/src/vim.md [hunk 1]
```diff
@@ -442,15 +442,15 @@ Here's an example of these settings changed:
...
+| Property                | Description                                                                                                                                                   | Default Value      |
+| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ |
+| cursor_blink            | If `true`, the cursor blinks.                                                                                                                                 | `true`             |
+| relative_line_numbers   | If `true`, line numbers in the left gutter are relative to the cursor.                                                                                        | `false`            |
+| scrollbar               | Object that controls the scrollbar display. Set to `{ "show": "never" }` to hide the scroll bar.                                                              | `{ "show": null }` |
```
Comment (notpeter) [inline docs/src/vim.md:449]: `scrollbar.show` defaults to `auto` I believe.
https://github.com/zed-industries/zed/blob/39fdb1efb7a227ca926c4d7dfae0e824d9e29a95/assets/settings/default.json#L245

The real default is actually actually:
```
{"show": "auto", "cursors": true, "git_diff": true, "search_results": true, "selected_symbol": true, "diagnostics": true}
```
But I think that would format poorly so I think `{ "show": "auto" }` would be fine.

Comment (jaxvanyang) [reply docs/src/vim.md]: Sorry, it's my mistake. I blindly take the default value of `project_panel.scrollbar`. I should have double-checked that.

**Comment by k4leg** [comment]
@jaxvanyang fix the default value for [calls.share_on_join](https://github.com/zed-industries/zed/blob/ebaa270bafbd8b4ca504436c5178c68dc81618e7/crates/call/src/call_settings.rs#L23) ([false](https://github.com/zed-industries/zed/blob/ebaa270bafbd8b4ca504436c5178c68dc81618e7/assets/settings/default.json#L220)), please.

**Comment by jaxvanyang** [comment]
> @jaxvanyang fix the default value for [calls.share_on_join](https://github.com/zed-industries/zed/blob/ebaa270bafbd8b4ca504436c5178c68dc81618e7/crates/call/src/call_settings.rs#L23) ([false](https://github.com/zed-industries/zed/blob/ebaa270bafbd8b4ca504436c5178c68dc81618e7/assets/settings/default.json#L220)), please.

Thanks for pointing out. But this PR was to fix typos in the doc, and I think the better way is deleting those comments. Maintaining simple comments adds extra work and makes this out-of-sync situation possible. After reading the Git history, I find that comments were deleted before, but it's reverted: https://github.com/noaccOS/zed/commit/e2fd12b1db6cf6a79b73de8a4c20703cbcec80c0#diff-dd075c60d48086283eeee0762d031e8dcab290911a3951a86c2178b950d21adeL11-R26

So I think deleting them would be OK. cc @osiewicz

**Comment by osiewicz** [comment]
Comments on structs that derive JsonSchema are usually there to show up in completions menu for `settings.json`. my PR that got reverted moved the defaults to programmatic definitions (that also show up in completions), but we've had issues integrating them. I'd say we should keep the comments for the time being; that said, your point about them adding extra work and possibly being out of sync is very valid and that is why I've wanted to change that in the first place. :)

**Comment by jaxvanyang** [comment]
> Comments on structs that derive JsonSchema are usually there to show up in completions menu for `settings.json`. my PR that got reverted moved the defaults to programmatic definitions (that also show up in completions), but we've had issues integrating them. I'd say we should keep the comments for the time being; that said, your point about them adding extra work and possibly being out of sync is very valid and that is why I've wanted to change that in the first place. :)

I see, that sounds like a huge work. Should I fix that value in this PR, or leave it to another integration?

**Review by maxdeviant — APPROVED** [review]
Thank you!

**Pull request merged**